Remove string processing done via shell and #486

horcsinbalint opened the pull request:
Using shell scripts to transform text contributes quite a bit to the smell of the codebase. We also had issues because of this (`grep` arbitrarily decided sometimes for outputs of `pdfinfo` that they were binary data and not text). While removing calls to `Process::fromShellCommandline` (which is quite unsafe), I also moved from `ping` to `nmap` (fixing https://github.com/EotvosCollegium/mars/issues/486).

horcsinbalint committed:
Remove string processing done via shell and #486

Using shell scripts to transform text contributes quite a bit to the
smell of the codebase. We also had issues because of this (`grep`
arbitrarily decided sometimes for outputs of `pdfinfo` that they were
binary data and not text). While removing calls to
`Process::fromShellCommandline` (which is quite unsafe), I also moved
from `ping` to `nmap` (fixing #486).

BertalanD commented on app/Utils/PrinterHelper.php:
```diff
@@ -19,7 +19,7 @@ class PrinterHelper
     public static function getDocumentPageNumber(string $path): int
     {
-        $process = Process::fromShellCommandline(config('commands.pdfinfo') . " $path | grep '^Pages' | awk '{print $2}'");
+        $process = Process::fromShellCommandline(config('commands.pdfinfo') . " $path | grep --text '^Pages' | awk '{print $2}'");
```
What. Why can't we just simply parse the command's output from within PHP. Opening a shell for this is very fishy.

horcsinbalint replied:
Thanks for the suggestion, I think I have remedied the issue (and some similar as well)

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in e7a36ea according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/726

BertalanD reviewed (commented):
Thank you, LGTM! (Assumimg this has been tested)
